### PR TITLE
proposal:  pause ensure work object

### PR DESCRIPTION
**Pull request by CharlesQQ**
Signed-off-by: charlesQQ <charles_ali@qq.com>

**What type of PR is this?**
/kind design
/kind documentation

**What this PR does / why we need it**:

pause ensure work object

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/1567

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```
None

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
This week will be pretty rough for me as we are planning to cut the new release, I'm not sure I can take a look at this approach. But I've added this to the [release-1.3 backlog](https://github.com/orgs/karmada-io/projects/3).

File: docs/proposals/pause-ensure-work/README.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+creation-date: 2022-05-21
+
+---
+
+# Pause ensure work
+
+## Summary
+
+When the resource (as known as `resource template`) paused field(such as label `work.karmada.io/paused`)  is true, we hope that the latest change, whether it is workload or PP or OP, will not be updated to the work object until it becomes false
```
Comment (RainbowMango) [inline docs/proposals/pause-ensure-work/README.md:17]: The summary part should describe what the proposal is. e.g. Do you propose a `work.karmada.io/paused` label?

File: docs/proposals/pause-ensure-work/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+## Motivation
+Users might add/replace/delete overriders for OP dynamically, and add/remove targetCluster in PP dynamically;
+These operations should be transactional, before all modifications are done, policy should not take effect
```
Comment (RainbowMango) [inline docs/proposals/pause-ensure-work/README.md:21]: The `motivation` part focuses on the reason why raise the proposal and why people need it.

File: docs/proposals/pause-ensure-work/README.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+creation-date: 2022-05-21
+
+---
+
+# Pause ensure work
+
+## Summary
+Karmada should provide an ability to allow user to decide whether sync resource (as known as `resource template`) to member cluster or not; 
+eg: When the resource  paused field(such as label `work.karmada.io/paused`)  is true, we hope that resource's not be updated member cluster until it becomes false
```
Comment (ikaven1024) [inline docs/proposals/pause-ensure-work/README.md:17]:
```suggestion
eg: When the resource's paused field(such as label `work.karmada.io/paused`)  is true, we hope that resource's not be updated member cluster until it becomes false
```

**Comment by RainbowMango** [comment]
I'm trying to figure out which PR/Issue should be included in the coming v1.7 release which is planned at the end of this month.
I guess we don't have enough time for this feature, so I'm moving this to v1.8.

**Comment by CharlesQQ** [comment]
https://github.com/karmada-io/karmada/issues/4421

**Comment by XiShanYongYe-Chang** [comment]
#5217 fix this pr.
Hi @CharlesQQ, can we close this pr?

**Pull request closed without merging**